exporter/coralogixexporter/traces_client.go

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coralogixexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/coralogixexporter"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/pdata/ptrace/ptraceotlp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type tracesExporter struct {
+	// Input configuration.
+	config *Config
+
+	traceExporter ptraceotlp.Client
+	clientConn    *grpc.ClientConn
+	metadata      metadata.MD
+	callOptions   []grpc.CallOption
+
+	settings component.TelemetrySettings
+
+	// Default user-agent header.
+	userAgent string
+}
+
+func newTracesExporter(cfg config.Exporter, set component.ExporterCreateSettings) (*tracesExporter, error) {
+	oCfg := cfg.(*Config)
+
+	if oCfg.Endpoint == "" || oCfg.Endpoint == "https://" || oCfg.Endpoint == "http://" {
+		return nil, errors.New("coralogix exporter config requires `endpoint` configuration")
+	}
+	userAgent := fmt.Sprintf("%s/%s (%s/%s)",
+		set.BuildInfo.Description, set.BuildInfo.Version, runtime.GOOS, runtime.GOARCH)
+
+	return &tracesExporter{config: oCfg, settings: set.TelemetrySettings, userAgent: userAgent}, nil
+}
+
+func (e *tracesExporter) start(_ context.Context, host component.Host) (err error) {
```

Comment:
```suggestion
func (e *tracesExporter) start(_ context.Context, host component.Host) error {
```
Is there a reason to use a named return statement here? Maybe we should avoid it to be consistent with other impl. parts.